airflow/jobs/scheduler_job.py

```diff
@@ -236,22 +236,19 @@ def done(self):
         if self._done:
             return True
 
-        # In case result queue is corrupted.
-        if self._result_queue and not self._result_queue.empty():
-            self._result = self._result_queue.get()
+        if self._parent_channel.poll():
+            self._result = self._parent_channel.recv()
             self._done = True
             self.log.debug("Waiting for %s", self._process)
             self._process.join()
+            self._parent_channel.close()
             return True
 
-        # Potential error case when process dies
-        if self._result_queue and not self._process.is_alive():
+        if not self._process.is_alive():
             self._done = True
-            # Get the object from the queue or else join() can hang.
-            if not self._result_queue.empty():
-                self._result = self._result_queue.get()
             self.log.debug("Waiting for %s", self._process)
             self._process.join()
```

Comment:
I think we do not have to join() after we found that _process.is_alive() is False. But it is no harm and it's the kind of sanity way of doing it.

Reply:
See next comment.